File: Review-00-Python-Head-to-Head-Testing/Example-2/even_gen.py
import sys


def output_even_integers(lower: int, upper: int) -> None:
    """
    Output all even numbers in a given range.

    # Args

      - lower lower integer bound (a)
      - upper upper integer bound (b)
    """

    for next_even in range(lower, upper + 1, 2):
        print(f"{next_even}")


def main():
    # Check and parse command line args
    if len(sys.argv) < 3:
        print(" Usage: ./even_gen [lower_bound] [upper_bound]")
        sys.exit(1)

    # Assume all args are well formed (i.e., can be parsed as integers).
    lower_bound = int(sys.argv[1])
    upper_bound = int(sys.argv[2])

    # The core even output logic
    print(f"Range [{lower_bound}, {upper_bound}]")
    print()

    output_even_integers(lower_bound, upper_bound)


if __name__ == "__main__":
    main()

File: 00-Java-Head-to-Head-Testing/Example-1/src/main/java/EvenGen.java

public class EvenGen
{
    public static void main(String... args)
    {
        if (args.length < 2) {
            System.out.println(" Usage: ./even_gen [lower_bound] [upper_bound]");
            System.exit(1);
        }

        // Assume all args are well formed (i.e., can be parsed as integers).
        final int lowerBound = Integer.parseInt(args[0]);
        final int upperBound = Integer.parseInt(args[1]);

        // The core even output logic
        System.out.printf("Range [%d, %d]", lowerBound, upperBound);
        System.out.println();

        for (int nextEven = lowerBound; nextEven <= upperBound; nextEven++) {
            System.out.printf("%d%n", nextEven);
        }

    }
}
